src/Invariants/Equality.jl

```diff
@@ -0,0 +1,43 @@
+using Graphs
+
+"""
+equals method for undirected edges::Union{Vector{Tuple{Int, Int}}, Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}}
+"""
+function equals(
+    a::Union{Vector{Tuple{Int, Int}}, Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}},
+    b::Union{Vector{Tuple{Int, Int}}, Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}}
+)
+
+    # convert a to an array of tuples
+    if typeof(a) == Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}
+        a = [(min(e.src, e.dst), max(e.src, e.dst)) for e in a]
+    else
+        a = [(min(e[1], e[2]), max(e[1], e[2])) for e in a]
+    end
+
+    # convert b to an array of tuples
+    if typeof(b) == Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}
+        b = [(min(e.src, e.dst), max(e.src, e.dst)) for e in b]
+    else
+        b = [(min(e[1], e[2]), max(e[1], e[2])) for e in b]
+    end
+
+    # check if the lengths are equal
+    if length(a) != length(b)
```

Comment:
A more Julia way of doing this would be with the following:

```julia 
length(a) != length(b) && return false 
```

Reply:
Is this equivalent to
`length(a) == length(b) || return false`
and if so is there a preference?

Reply:
Use DeMorgans law for an equivalence

Reply:
DeMorgan's Law gives: 

```julia 
length(a) == length(b) || return true
```

as an equivalence

Reply:
Also, there is no preference. I just typically use `&&` when working in julia

Reply:
I'm not sure I understand how to apply DeMorgans law to short circuit logic. This is the way I see it:
We want to short circuit when the lengths are the same and to return false otherwise.

so in one case `length(a) != length(b) && return false` short circuits when `length(a) != length(b)` is not true or when the lengths are the same. Otherwise, return false.

In the case I offered, `length(a) == length(b) || return false` short circuits if the first condition is true; that is once again when the lengths are the same. Otherwise, return false.

Have I made a mistake?

Reply:
Seeing your example, I think I see what you mean: `length(a) == length(b) || return true` is the negation of the original statement, so we can flip the return back to false to get a statement of equivalent function?

Reply:
Also I am starting to see why `&&` is more natural even if the first statement is a negative. It reads better as the if-then control flow we are aiming for